Packs/OracleCloudInfrastructure/Integrations/OracleCloudInfrastructureEventCollector/OracleCloudInfrastructureEventCollector.py

```diff
@@ -0,0 +1,345 @@
+        Returns:
+            List[Dict[str, Any]]: The events with the _time key.
+        """
+        for event in events:
+            if event.get("eventTime"):
+                event["_time"] = event.get("eventTime")
+
+        return events
+
+
+def test_module(client: Client, oci_event_handler: OCIEventHandler) -> str:
+    """
+    Tests API connectivity and authentication.
+    When 'ok' is returned it indicates the integration works like it is supposed to and connection to the service is
+    successful.
+    Raises exceptions if something goes wrong.
+    Args:
+        client (Client): Client for SDK interaction and api requests.
+        oci_event_handler (OCIEventHandler): OCI event handler object.
+    Returns:
+        str: 'ok' if test passed, anything else will raise an exception and will fail the test.
+    """
+    try:
+        params = {
+            'compartmentId': oci_event_handler.client.compartment_id,
+            'startTime': datetime.now().isoformat(),
+            'endTime': datetime.now().isoformat()
+        }
+        client._http_request(method='GET', params=params)
+
+    except Exception as e:
+        if 'failed' in str(e):
+            return 'Authorization Error: make sure OCI parameters are correctly set'
+        else:
+            raise DemistoException(f'Error while testing: {e}') from e
+
+    return 'ok'
+
+
+def calculate_first_fetch_time(last_run: Optional[str], first_fetch_param: str) -> Optional[datetime]:
+    """Calculates the first fetch time.
+
+    Args:
+        last_run (Optional[str]): Last run time from previous fetch.
+        first_fetch_param (str): First Fetch Time integration parameter.
+
+    Returns:
+        Optional[datetime]: Maximum datetime value between last run from previous fetch and first fetch time parameter.
+    """
+    first_fetch_param_datetime = arg_to_datetime(arg=first_fetch_param)
+
+    # if last_run is None (first time we are fetching) -> return first_fetch_arg datetime object
+    if not last_run:
+        return first_fetch_param_datetime
+    else:
+        last_run_datetime = arg_to_datetime(arg=last_run, settings={'RETURN_AS_TIMEZONE_AWARE': False})
+
+    # if last_run is not None -> return max(last_run, first_fetch_arg)
+    if last_run_datetime and first_fetch_param_datetime:
+        return max(last_run_datetime, first_fetch_param_datetime)
+    else:  # return default first fetch time datetime object
+        return arg_to_datetime(arg=FETCH_DEFAULT_TIME)
+
+
+''' MAIN FUNCTION '''
+
+
+def main():
+    """
+    main function, parses params and runs command functions
+    """
+
+    params = demisto.params()
+    args = demisto.args()
+    command = demisto.command()
+    last_run = demisto.getLastRun()
+    last_run_time = last_run.get('lastRun')
+    demisto.info(f'last_run_time value {last_run_time}')
+    max_fetch = arg_to_number(params.get('max_fetch')) or MAX_EVENTS_TO_FETCH
+    first_fetch = params.get('first_fetch') or FETCH_DEFAULT_TIME
```

Comment:
Why do we use 'or' statement and not a default value for the get func? 
Is it due to mypy error?

Reply:
`arg_to_number` can return a None value. using or insures that `max_fetch` is intiliazed with a number value and not None.